File: app/src/main/java/com/example/terry/ssumap_android/MainActivity.java
package com.example.terry.ssumap_android;

import android.Manifest;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.Toast;

import com.gun0912.tedpermission.PermissionListener;
import com.gun0912.tedpermission.TedPermission;

import net.daum.mf.map.api.MapView;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    private Button btnFacility;
    private Button btnTour;
    private Button btnCurLocation;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btnFacility = findViewById(R.id.btn_facility);
        btnTour = findViewById(R.id.btn_tour);
        btnCurLocation = findViewById(R.id.btn_cur_loc);

        btnFacility.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(MainActivity.this, FacilityListActivity.class));
            }
        });

        PermissionListener permissionlistener = new PermissionListener() {
            @Override
            public void onPermissionGranted() {
                MapView mapView = new MapView(MainActivity.this);
                mapView.setDaumMapApiKey("ea8d49685012f33f11397e22230ecba1");

                ViewGroup mapViewContainer = (ViewGroup) findViewById(R.id.map_view);
                mapViewContainer.addView(mapView);
            }

            @Override
            public void onPermissionDenied(ArrayList<String> deniedPermissions) {

            }
        };

        TedPermission.with(this)
                .setPermissionListener(permissionlistener)
                .setRationaleMessage("서비스 이용을 위해서는 위치 권한이 필요합니다.")
                .setDeniedMessage("[설정] > [권한] 에서 권한을 허용해 주세요.")
                .setPermissions(Manifest.permission.ACCESS_FINE_LOCATION)
                .check();
    }

}

File: app/src/main/java/com/example/terry/ssumap_android/MainActivity.kt
package com.example.terry.ssumap_android

import android.content.Context
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.os.Handler
import android.widget.Toast

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        withDelay(2000) {
            functionX()
        }
    }

    // Temporary Functions
    fun functionX() {
        "After 2000 MilliSeconds".toast(this)
    }

    // Extension Functions
    fun withDelay(delay : Long, block : () -> Unit) {
        Handler().postDelayed(Runnable(block), delay)
    }

    // Extension Function for show Toast to Device
    fun Any.toast(context: Context) {
        Toast.makeText(context, this.toString(), Toast.LENGTH_LONG).show()
    }
}
